openff/interchange/_tests/unit_tests/smirnoff/test_nonbonded.py

```diff
@@ -135,6 +135,347 @@ def test_toolkit_am1bcc_uses_elf10_if_oe_is_available(self, sage, hexane_diol):
+        ff.deregister_parameter_handler("ToolkitAM1BCC")
+
+        interchange = Interchange.from_smirnoff(force_field=ff, topology=hexane_diol.to_topology())
+        assigned_charges = interchange["Electrostatics"].get_charge_array()
+
+        # Should match NAGL charges, not formal charges
+        numpy.testing.assert_allclose(assigned_charges, nagl_charges)
+
+
+class TestNAGLChargesIntegration:
+    """Test NAGLCharges integration with other handlers."""
+
+    def test_nagl_charges_multi_molecule_topology(self):
+        """Test NAGLCharges with multiple molecules in topology."""
+        from openff.toolkit.typing.engines.smirnoff import ForceField
+
+        from openff.interchange import Interchange
+
+        methane = Molecule.from_smiles("C")
+        ethane = Molecule.from_smiles("CC")
+
+        topology = Topology.from_molecules([methane, ethane])
+
+        ff = ForceField("openff-2.1.0.offxml")
+        ff.get_parameter_handler(
+            "NAGLCharges",
+            {
+                "model_file": "openff-gnn-am1bcc-0.1.0-rc.3.pt",
+                "version": "0.3",
+            },
+        )
+
+        interchange = Interchange.from_smirnoff(force_field=ff, topology=topology)
+        assigned_charges = interchange["Electrostatics"].get_charge_array()
+
+        # Should have charges for all atoms
+        assert len(assigned_charges) == topology.n_atoms
+
+        # Each molecule should have approximately zero net charge
+        methane_charge_sum = sum(assigned_charges[: methane.n_atoms])
+        ethane_charge_sum = sum(assigned_charges[methane.n_atoms :])
+
+        assert abs(methane_charge_sum) < 1e-6 * unit.elementary_charge
+        assert abs(ethane_charge_sum) < 1e-6 * unit.elementary_charge
+
+    def test_nagl_charges_with_virtual_sites(self, sage_with_bond_charge):
+        """Test NAGLCharges compatibility with virtual sites."""
+        from openff.interchange import Interchange
+
+        # Create a molecule that would have virtual sites
+        molecule = Molecule.from_smiles("[Cl]CCO")
+
+        # Add NAGLCharges to the force field
+        sage_with_bond_charge.get_parameter_handler(
+            "NAGLCharges",
+            {
+                "model_file": "openff-gnn-am1bcc-0.1.0-rc.3.pt",
+                "version": "0.3",
+            },
+        )
+
+        # Should not raise an error
+        interchange = Interchange.from_smirnoff(
+            force_field=sage_with_bond_charge,
+            topology=molecule.to_topology(),
+        )
+
+        # Should have charges for real atoms
+        assigned_charges = interchange["Electrostatics"]._get_charges()
+        assert len(assigned_charges.values()) - 1 == molecule.n_atoms
+
+        # Net charge should be approximately zero
+        all_particle_charge_sum = sum(assigned_charges.values())
+        assert abs(all_particle_charge_sum) < 1e-6 * unit.elementary_charge
```

Comment:
A bit out of scope here, but I'd hope we get closer to 0 than 1e-6? That seems fine for a single molecule but accumulates in large systems